base/base.jl

```diff
@@ -34,6 +34,11 @@ type MethodError <: Exception
     args
 end
 
+type AmbiguousError <: Exception
```

Comment:
Well, it's not the error that's ambiguous anyway, so both
AmbiguousMethodError and AmbiguityError seem better than AmbiguousError.